numpy/testing/decorators.py

```diff
@@ -3,4 +3,9 @@
 set of tools
 
 """
-from .nose_tools.decorators import *
+import os
+
+if int(os.getenv('NPY_PYTEST', '0')):
```

Comment:
Is it a good idea to make module contents depend on environment variables?

Would it not be better to tell users to import stuff either from
`numpy.testing.nose_tools` or `numpy.testing.pytest_tools`, depending on which test framework they are using?

Reply:
If it were up to me, I would make the default `pytest` if it can be imported. I'm not going to change this until I hear from @charris though because this is his work.

Reply:
That would be a backward compat break, which probably cannot be made.

Reply:
Matplotlib had a similar problem.  We still ship all of the nose tooling even if we do not use it.

Reply:
I expect everyone downstream to be importing from `np.testing`, so asking them to import from the separate packages could be a problem. I tried to set things up so that it was transparent, especially if folks are testing against older versions of NumPy where the new directories are not present. I suppose another option would be to leave things as is for Nose testing and ask folks switching to PyTest to import directly from the lower level directory.

Reply:
However, we can keep things as is until NumPy is switched over.

Reply:
Honestly that's probably the better option. Tests aren't usually written in an interactive console, so a short import path isn't worth much, and that's weighed against giving downstream projects a very unpleasant nightmare. I will remove this.

Reply:
@xoviat It breaks backwards compatibility downstream.

Reply:
There will be no backwards compatibility breaks for the `numpy.testing` namespace in this PR. I need some time to figure out what needs to be patched though.